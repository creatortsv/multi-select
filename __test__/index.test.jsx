import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MultiSelect from '../src';

configure({ adapter: new Adapter() });

test('MultiSelect does not change list of selected and fixed options after term changed', () => {
  const options = [{
    label: 'group',
    children: [{
      value: 1,
      label: 'option',
    }, {
      value: 2,
      label: 'option 2',
    }],
  }, {
    value: 3,
    label: 'option 3',
  }];

  const select = shallow(
    <MultiSelect value={[3]} options={options} fixed={[1]} />
  );

  expect(shallowToJson(select)).toMatchSnapshot();

  select.find('input').simulate('change', {
    currentTarget: {
      value: 'text',
    },
    preventDefault() {},
  });

  expect(shallowToJson(select)).toMatchSnapshot();
});
