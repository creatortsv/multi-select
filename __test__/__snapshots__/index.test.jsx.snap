// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect does not change list of selected and fixed options after term changed 1`] = `
<div
  className="ms--wrapper"
  style={Object {}}
>
  <select
    disabled={false}
    multiple={true}
    onChange={[Function]}
    value={
      Array [
        3,
      ]
    }
  >
    <Option
      key="0"
      label="option"
      value={1}
    />
    <Option
      key="1"
      label="option 2"
      value={2}
    />
    <Option
      key="2"
      label="option 3"
      value={3}
    />
  </select>
  <div
    className="ms--container"
    style={Object {}}
  >
    <div
      className="ms--section"
      style={Object {}}
    >
      <div>
        <input
          className=""
          onChange={[Function]}
          placeholder="All elements"
          style={
            Object {
              "display": "block",
              "width": "100%",
            }
          }
          value=""
        />
      </div>
      <Group
        className="ms--group"
        disabled={false}
        key="0"
        label="group"
        onSelect={[Function]}
        open={false}
        openClass="ms--group__open"
        openStyle={Object {}}
        style={Object {}}
        values={
          Array [
            1,
            2,
          ]
        }
      >
        <Component
          active={false}
          className="ms--item"
          key="1"
          label="option 2"
          onSelect={[Function]}
          pinned={false}
          style={Object {}}
          value={2}
        />
      </Group>
      <div>
        <a
          href="#"
          onClick={[Function]}
        >
          Select all
        </a>
      </div>
    </div>
    <div
      className="ms--section"
      style={Object {}}
    >
      <p>
        Selected elements
      </p>
      <Component
        active={true}
        className="ms--item"
        key="1"
        label="option 3"
        onSelect={[Function]}
        pinned={false}
        style={Object {}}
        value={3}
      />
      <div>
        <a
          href="#"
          onClick={[Function]}
        >
          Unselect all
        </a>
      </div>
    </div>
    <div
      className="ms--section"
      style={Object {}}
    >
      <p>
        Fixed elements
      </p>
      <Component
        active={false}
        className="ms--item"
        key="0"
        label="option"
        pinned={true}
        style={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`MultiSelect does not change list of selected and fixed options after term changed 2`] = `
<div
  className="ms--wrapper"
  style={Object {}}
>
  <select
    disabled={false}
    multiple={true}
    onChange={[Function]}
    value={
      Array [
        3,
      ]
    }
  >
    <Option
      key="0"
      label="option"
      value={1}
    />
    <Option
      key="1"
      label="option 2"
      value={2}
    />
    <Option
      key="2"
      label="option 3"
      value={3}
    />
  </select>
  <div
    className="ms--container"
    style={Object {}}
  >
    <div
      className="ms--section"
      style={Object {}}
    >
      <div>
        <input
          className=""
          onChange={[Function]}
          placeholder="All elements"
          style={
            Object {
              "display": "block",
              "width": "100%",
            }
          }
          value="text"
        />
      </div>
      <Group
        className="ms--group"
        disabled={true}
        key="0"
        label="group"
        onSelect={[Function]}
        open={false}
        openClass="ms--group__open"
        openStyle={Object {}}
        style={Object {}}
        values={Array []}
      />
      <div>
        <a
          href="#"
          onClick={[Function]}
        >
          Select all
        </a>
      </div>
    </div>
    <div
      className="ms--section"
      style={Object {}}
    >
      <p>
        Selected elements
      </p>
      <Component
        active={true}
        className="ms--item"
        key="1"
        label="option 3"
        onSelect={[Function]}
        pinned={false}
        style={Object {}}
        value={3}
      />
      <div>
        <a
          href="#"
          onClick={[Function]}
        >
          Unselect all
        </a>
      </div>
    </div>
    <div
      className="ms--section"
      style={Object {}}
    >
      <p>
        Fixed elements
      </p>
      <Component
        active={false}
        className="ms--item"
        key="0"
        label="option"
        pinned={true}
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
